### ISSUE TITLE
`lines-around-comment`: `allowParameterStart`/`allowParameterEnd` needed

### ISSUE THREAD

**Issue by essenmitsosse** [eslint-stylistic/eslint-stylistic#462]
### Clear and concise description of the problem

Currently there is no way to allow to comments to appear at the start of parameter lists, similar to how `allowBlockStart`, `allowObjectStart` etc. work.

Currently `beforeBlockComment` enforces a styling like this:
```js
const myFunc = (

    /** Comment describing the first parameter */
    firstParam,

    /** Comment describing the second parameter */
    secondParam,
) => {}
```

Where the new line before the first parameter is unnecessary. For objects we can fix this with `allowObjectStart`:
```js
const myObj = {
    /** Comment describing the first property */
    firstProp,
   
    /** Comment describing the second property */
    secondProp,
}
```

### Suggested solution

In the rule `lines-around-comment` we could add options for `allowParameterStart` and `allowParameterEnd`, that allow comments to appear at the start (or end) of parameter lists.

### Additional context

Currently the lack of this feature also makes it incompatible with Prettier.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue
